### ISSUE TITLE
Revision Ideas to Streamline DH Class Coding Sequences

### ISSUE THREAD

**Issue by ebeshero** [ebeshero/newtFire-webDev#105]
@jonhoranic @ghbondar @quantum-satire Here's an issue to get us senior-most members of the DH instructor team to think about revisions to the course to make it run more smoothly and be encouraging /motivational and less flummoxing to our students.  Please post your ideas here. 

One big idea I had was to shorten the required length of many homework assignments, so that students currently don't have to finish all the questions, but finish half, submit, and we do the rest in class. Follow this up for each unit with a take-home exam worth more points than the homeworks (and longer) to prove that the students have grokked the unit successfully.

The next idea was to break up the XPath unit into pieces, with easy XPath first, followed immediately by some easy XSLT or Schematron to apply it. Then add complexity (predicates / functions / string-matching functions) to XPath, and do a corresponding application to show how to process this. 

I'd like us here to identify homework sequences or individual exercises that seem problematic / too hard / time consuming that really need streamlining/revision. You can be merciless about this: some of our assignments were written in a hurry and have too much text! (See some of the XSLT). How and where can we be revising and cutting back? Post links and comments here, and we'll work on this stuff over the next week as we're getting ready for our fall class. 

Thanks!

**Comment by ebeshero** [comment]
@djbpitt @zme1 @gabikeane  I'm extending a special invitation for you obdurodonians (and honorary newtonians) to assist us with this issue, only if you wish! :-) A key aspect of this issue is to help us identify newtfire assignments and tutorials that could simply be better written (which is probably all of them, sure), but some are more egregiously lengthy or potentially unclear than others, and we'd like to prioritize which ones need the most immediate work. We can start here: http://dh.newtfire.org/ to see a complete list, and let's concentrate on the materials marked for the fall CDA class. 

Post links to your least favorite exercises and comment briefly on what you think is problematic / how we could change it.

**Comment by gabikeane** [comment]
I love the idea of beginning XPath and then pausing to implement it in XSLT especially-- I will certainly take a look at the site in the next week and post here then.

**Comment by gabikeane** [comment]
I'm in the middle of doing a more thorough review, but this one seems rather pressing. The Regex test is still up! Ack!

**Comment by ebeshero** [comment]
Thanks, @gabikeane -- It's probably not a big deal, though, since it's essentially going to be a take-home exercise anyway and subject to modification as we're rethinking what an exam consists of. The answers aren't posted as far as I can see.

**Comment by helvitiis** [comment]
@ebeshero Are the DH pages down? I started combing through a few days ago and when I resumed today I can't seem to access them!

**Comment by ebeshero** [comment]
@quantum-satire Yes and no--they may be running, um, intermittently: I'm working on migrating to a  new server host right now, and I'll be stabilizing things shortly. I'll need you all to help test it--just a wait a bit for me to complete operations...

**Comment by helvitiis** [comment]
@ebeshero Sounds good! Glad nothing catastrophic happened, I was worried for a minute...
